code/modules/antagonists/creep/creep.dm

```diff
@@ -149,7 +149,7 @@
 /datum/objective/assassinate/obsessed/update_explanation_text()
 	..()
 	if(target && target.current)
-		explanation_text = "Murder [target.name], the [!target_role_type ? target.assigned_role : target.special_role]."
+		explanation_text = "Murder [target.name], the [!target_role_type ? target.get_station_role() : target.get_special_role()]."
```

Comment:
bad explain text! No metagaming! smh

Reply:
all things you're talking about is realted with target_role_type variable